lxd/storage/backend_lxd.go

```diff
@@ -6952,6 +6952,58 @@ func (b *lxdBackend) createStorageStructure(path string) error {
 	return nil
 }
 
+// UpdateCustomVolumeBackupFile writes the custom volume's config to the backup.yaml file of the corresponding instances.
+// A list of instances can be provided to not require looking them up.
+func (b *lxdBackend) UpdateCustomVolumeBackupFile(projectName string, volName string, snapshots bool, instances []instance.Instance, op *operations.Operation) error {
+	l := b.logger.AddContext(logger.Ctx{"project": projectName, "volume": volName, "snapshots": snapshots})
+	l.Debug("UpdateCustomVolumeBackupFile started")
+	defer l.Debug("UpdateCustomVolumeBackupFile finished")
+
+	vol, err := VolumeDBGet(b, projectName, volName, drivers.VolumeTypeCustom)
+	if err != nil {
+		return err
+	}
+
+	// Fetch all instances which are currently using the custom volume in one of their devices if not provided already.
+	if instances == nil {
+		err = VolumeUsedByInstanceDevices(b.state, b.name, projectName, &vol.StorageVolume, true, func(dbInst db.InstanceArgs, project api.Project, _ []string) error {
+			inst, err := instance.Load(b.state, dbInst, project)
+			if err != nil {
+				return err
+			}
+
+			instances = append(instances, inst)
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	// Fetch the latest custom volume backup config once for all instances.
+	volConfig, err := b.GenerateCustomVolumeBackupConfig(projectName, volName, true, op)
+	if err != nil {
+		return fmt.Errorf("Failed generating volume backup config: %w", err)
+	}
+
+	// Update the backup config file of all instances.
+	// Try updating all of the instance's backup files using a best effort strategy.
+	// In case a custom volume is used by many instances, it might happen that an instance gets deleted
+	// whilst this function tries to propagate an update of a custom volume.
+	// A lock isn't acquired in this case for all the instances as this might cause too much interruption
+	// as every update of an instance's backup file requires mounting the corresponding volume.
+	for _, inst := range instances {
+		// Don't invoke the the update of the instance's backup file directly on *lxdBackend.
+		// The instance's root vol might be located on a different storage pool which gets loaded when calling the instance's UpdateBackupFile.
+		err = inst.UpdateBackupFile(volConfig)
```

Comment:
this looks wrong to me, you're getting the single update volume's config and passing it to `inst.UpdateBackupFile` which in turn will pass it to `pool.UpdateInstanceBackupFile` which in turn will pass it to `GenerateInstanceBackupConfig` which means it wont lookup the config for any other attached volumes on the instance (https://github.com/canonical/lxd/pull/15523/files#diff-029fdd0e1afdacb0e66630c0d7e813772122db9a7d036060902838db315d9c4dR7228) and will remove them from the backup.yaml file.